compiler/rustc_errors/src/diagnostic_impls.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{fluent, DiagnosticBuilder, Handler, IntoDiagnostic};
+use rustc_target::abi::TargetDataLayoutErrors;
+
+impl IntoDiagnostic<'_, !> for TargetDataLayoutErrors<'_> {
```

Comment:
If we're going to create a `diagnostic_impls` module, could you move the other pre-existing impls into it?

Reply:
Addressed [here](https://github.com/rust-lang/rust/pull/102684/commits/be221573c853f5cdca8becab5acc600d6c79eac5)